scripts/operations/configuration/python_lib/api_requests.py

```diff
@@ -31,6 +33,17 @@
 import requests
 
 from . import common
+from . import k8s
+from .types import JSONDecodeError
+
+
+ADMIN_CLIENT_SECRET_NAME = "admin-client-auth"
+ADMIN_CLIENT_SECRET_NAMESPACE = "default"
+AUTH_TOKEN_URL = ("https://api-gw-service-nmn.local/"
+                  "keycloak/realms/shasta/protocol/openid-connect/token")
+
+# For type hints
+ApiResponse = requests.models.Response
 
 
 def log_error_raise_exception(msg: str, parent_exception: Exception = None) -> None:
```

Comment:
Is there a reason this function is defined in every module? Can't it just be defined once in `common.py`?

Reply:
Actually, I'm glad you reminded me, because I wanted to ask you about this. The reason I define it locally is because when logging calls are made, they record the source file generating the message. This is meant to be a helper function for logging, and I want it to record the current source file in the log entries, not common.py. Does that make sense? And can you think of an alternative way to accomplish that?

Reply:
Maybe it could take the module name as a parameter, e.g. something like this?

```
log_error_raise_exception(msg='We failed', parent_exception=exc, module=__name__)
```

I don't know, that seems kind of awkward. This gets back into the other conversations we've had about error handling as well as the conversation here: https://github.com/Cray-HPE/docs-csm/pull/2933#discussion_r1046438497

But I think normally would I would do in error handling cases is just something like:
```
try:
  something_that_raises_an_error()
except TheError as e:
  LOGGER.error('We got an error ... ')
  raise
```

Then something higher up would handle `TheError`.

I don't feel super strongly about addressing this as you have a good point here, but I suspect that reworking the error handling more fundamentally could help with some of these weird quirks.